### PR TITLE
call enrich on admin list API

### LINKED ISSUE #703: enricher is not called in admin api list

### PR DESCRIPTION
**Pull request by jim-minter**
this adds enrichment to admin API list and also makes the API pageable.  It'll need a corresponding change to the geneva action codebase.

fixes #703

### CODE REVIEW

File: pkg/util/clusterdata/clusterdata.go [hunk 1]
```diff
@@ -58,16 +58,38 @@ type bestEffortEnricher struct {
 }
 
 func (e *bestEffortEnricher) Enrich(ctx context.Context, ocs ...*api.OpenShiftCluster) {
```
Comment (jim-minter) [inline pkg/util/clusterdata/clusterdata.go:60]: the changes here allow enrich to receive an arbitrary number of `ocs` to enrich.  Up to 10 are enriched concurrently each with a 10s timeout.

**Comment by m1kola** [comment]
I will take a look tomorrow. Please don't merge before I look properly, if possible.

**Comment by m1kola** [comment]
Update to this: we had discussion with Jim last week. I think we decided:

* Do not make any changes to the enricher itself (don't want to make the complex thing even more complex)
* Add enricher to admin list and add pagination
* Implement pagination on the Geneva action side

**Comment by jim-minter** [comment]
+1 - moved back to draft for the interim

**Comment by jim-minter** [comment]
@m1kola please can you re-review this?

File: pkg/frontend/admin_openshiftcluster_list.go [hunk 2]
```diff
@@ -27,37 +25,12 @@ func (f *frontend) getAdminOpenShiftClusters(w http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	b, err := f._getAdminOpenShiftClusters(ctx, r, f.apis[admin.APIVersion].OpenShiftClusterConverter())
+	b, err := f._getOpenShiftClusters(ctx, r, f.apis[admin.APIVersion].OpenShiftClusterConverter(), func(skipToken string) (cosmosdb.OpenShiftClusterDocumentIterator, error) {
+		return f.db.OpenShiftClusters.List(skipToken), nil
```
Comment (m1kola) [inline pkg/frontend/admin_openshiftcluster_list.go:29]: I like that we removed a bunch of (almost) duplicated code.

**Comment by m1kola** [comment]
@jim-minter I just noticed that the PR is still in draft. Are you planning to add something to it? Or is it in draft so we don't merge before geneva action side is ready?

**Comment by jim-minter** [comment]
closing for #859

**Pull request closed without merging**